File: CSharpRewrite/CSharpRewrite/Assignments/K02.cs
using CSharpRewrite.Shared;

namespace CSharpRewrite.Assignments;

public class K02 : IQuestion
{
    public void Question1()
    {
        const int seisuu = 3;
        const float jissuu = 2.6f;
        const char moji = 'A';

        Console.WriteLine("変数seisuuの値は" + seisuu);
        Console.WriteLine("変数jissuuの値は" + jissuu);
        Console.WriteLine("変数mojiの値は" + moji);
    }

    public void Question2()
    {
        try
        {
            Console.Write("一つ目の整数は？");
            var a = int.Parse(Console.ReadLine() ?? "0");
            Console.Write("二つ目の整数は？");
            var b = int.Parse(Console.ReadLine() ?? "0");

            Console.WriteLine($"{a}÷{b}={a / b}...{a % b}");

        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question3()
    {
        try
        {
            Console.Write("一つ目の商品の値段は？");
            var priceA = int.Parse(Console.ReadLine() ?? "0");
            Console.Write("個数は？");
            var amountA = int.Parse(Console.ReadLine() ?? "0");
            Console.Write("二つ目の商品の値段は？");
            var priceB = int.Parse(Console.ReadLine() ?? "0");
            Console.Write("個数は？");
            var amountB = int.Parse(Console.ReadLine() ?? "0");

            Console.WriteLine($"お支払いは税込み￥{(int)((priceA * amountA + priceB * amountB) * 1.1f)}です");

        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question4()
    {
        try
        {
            var golem = new Golem(300, 80, 50);

            Console.WriteLine($"ゴーレム　（HP:{golem.Hp}　防御力：{golem.Defense}）\n");
            Console.WriteLine("HP：" + golem.Hp);
            Console.Write("今回の攻撃の値を入力してください＞");
            var atk = int.Parse(Console.ReadLine() ?? "0");
            atk = atk - golem.Defense > 0 ? atk - golem.Defense : 0;
            Console.WriteLine("ダメージは" + atk + "です");
            golem.Hp -= atk;
            Console.WriteLine("残りのHPは" + golem.Hp + "です");

        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }
}

File: ScalaRewrite/src/main/scala/assignments/K02.scala
package assignments

import shared.{Golem, Question}

import scala.io.StdIn

case class K02()

object K02 extends Question:
  override def question1(): Unit =
    val seisuu = 3
    val jissuu = 2.6f
    val moji = 'A'
    println(s"変数seisuuの値は$seisuu")
    println(s"変数jissuuの値は$jissuu")
    println(s"変数mojiの値は$moji")

  override def question2(): Unit =
    print("一つ目の整数は？")
    val number1 = StdIn.readInt()
    print("二つ目の整数は？")
    val number2 = StdIn.readInt()
    println(s"$number1 ÷ $number2 = ${number1 / number2}...${number1 % number2}")

  override def question3(): Unit =
    print("一つ目の商品の値段は？")
    val priceA = StdIn.readFloat()
    print("個数は？")
    val amountA = StdIn.readFloat()
    print("二つ目の商品の値段は？")
    val priceB = StdIn.readFloat()
    print("個数は？")
    val amountB = StdIn.readFloat()
    val total = (priceA * amountA + priceB * amountB) * 1.1f
    println(s"お支払いは税込み￥$total")

  override def question4(): Unit =
    val golem = Golem(300, 80, 50)
    println(s"ゴーレム　（HP：${golem.hp}　防御力：${golem.defense}）")
    println(s"HP：${golem.hp}")
    print("今回の攻撃の値を入力してください＞")
    val damage = StdIn.readInt()
    val actualDamage = if damage - golem.defense < 0 then 0 else damage - golem.defense
    println(s"ダメージは${actualDamage}です。")
    val newGolem = golem.copy(hp = golem.hp - actualDamage)
    println(s"残りのHPは${newGolem.hp}です。")